test/utils/fizzy_engine.cpp

```diff
@@ -0,0 +1,64 @@
+#include "execute.hpp"
+#include "parser.hpp"
+#include <test/utils/wasm_engine.hpp>
+#include <cassert>
+
+namespace fizzy
+{
+class FizzyEngine : public WasmEngine
+{
+    Instance m_instance;
+
+public:
+    std::string get_name() final;
+    void parse(bytes_view input) final;
+    std::optional<uint32_t> find_function(std::string_view name) const final;
+    void instantiate() final;
+    fizzy::bytes_view get_memory() const final;
+    void set_memory(fizzy::bytes_view memory) final;
+    execution_result2 execute(FuncIdx func_idx, const std::vector<uint64_t>& args) final;
+};
+
+std::unique_ptr<WasmEngine> create_fizzy_engine()
+{
+    return std::make_unique<FizzyEngine>();
+}
+
+std::string FizzyEngine::get_name()
+{
+    return "fizzy";
+}
+
+void FizzyEngine::parse(bytes_view input)
+{
+    m_instance.module = fizzy::parse(input);
+}
+
+void FizzyEngine::instantiate()
+{
+    m_instance = fizzy::instantiate(std::move(m_instance.module));
+}
+
+fizzy::bytes_view FizzyEngine::get_memory() const
+{
+    return {m_instance.memory.data(), m_instance.memory.size()};
+}
+
+void FizzyEngine::set_memory(fizzy::bytes_view memory)
+{
+    m_instance.memory = memory;
```

Comment:
You'll find that in many cases this will require an exported memory section from the benchmark to have the memory actually accessible in a VM.

I think we'll ensure that in `instantiate`, so have to complaints about the API.

Reply:
I don't understand what you mean here.

Reply:
Memory is not exported (made available) by wasm engines, unless the bytecode has an export memory statement. Fizzy just exposes things unconditionally currently.

Reply:
What's the problem though?

Reply:
Nothing with `FizzyEngine`.